src/test/java/cloud/fogbow/ras/core/OrderControllerTest.java

```diff
@@ -167,36 +146,271 @@ public void testGetInactiveOrder() throws InstanceNotFoundException {
+
+        // exercise
+        this.ordersController.activateOrder(order);
+        this.ordersController.deactivateOrder(order);
+
+        // verify
+        Assert.assertEquals(order.getOrderState(), OrderState.DEACTIVATED);
+    }
+
+    // test case: Creates an order with dependencies and check if the order id
+    // will be inserted into dependencies.
+    @Test
+    public void testCreateOrderWithDependencies() throws FogbowException {
+        // set up
+        ComputeOrder computeOrder = createLocalComputeOrder();
+        VolumeOrder volumeOrder1 = createLocalVolumeOrder();
+        VolumeOrder volumeOrder2 = createLocalVolumeOrder();
+
+        this.ordersController.activateOrder(computeOrder);
+        this.ordersController.activateOrder(volumeOrder1);
+        this.ordersController.activateOrder(volumeOrder2);
+
+        AttachmentOrder attachmentOrder1 = createLocalAttachmentOrder(computeOrder, volumeOrder1);
+        AttachmentOrder attachmentOrder2 = createLocalAttachmentOrder(computeOrder, volumeOrder2);
+
+        // exercise
+        this.ordersController.activateOrder(attachmentOrder1);
+        this.ordersController.activateOrder(attachmentOrder2);
+
+        // verify
+        Assert.assertTrue(this.orderHasDependencies(computeOrder.getId()));
+    }
+
+    // test case: Creates an order with dependencies and attempt to delete
+    // it must throw an DependencyDetectedException.
+    @Test(expected = DependencyDetectedException.class)
+    public void testDeleteOrderWithoutRemovingDependenciesFirst() throws FogbowException {
+        // set up
+        ComputeOrder computeOrder = createLocalComputeOrder();
+        VolumeOrder volumeOrder = createLocalVolumeOrder();
+
+        this.ordersController.activateOrder(computeOrder);
+        this.ordersController.activateOrder(volumeOrder);
+
+        AttachmentOrder attachmentOrder = createLocalAttachmentOrder(computeOrder, volumeOrder);
+        this.ordersController.activateOrder(attachmentOrder);
+
+        // exercise / verify
+        this.ordersController.deleteOrder(volumeOrder);
+    }
+
+    // test case: Creates an order with dependencies and attempt to delete
+    // them in correct order must not throw any exceptions.
+    @Test
+    public void testDeleteOrderWithDependencies() throws FogbowException {
+        // set up
+        ComputeOrder computeOrder = createLocalComputeOrder();
+        VolumeOrder volumeOrder = createLocalVolumeOrder();
+
+        this.ordersController.activateOrder(computeOrder);
+        this.ordersController.activateOrder(volumeOrder);
+
+        AttachmentOrder attachmentOrder = createLocalAttachmentOrder(computeOrder, volumeOrder);
+        this.ordersController.activateOrder(attachmentOrder);
+
+        // exercise / verify
+        this.ordersController.deleteOrder(attachmentOrder);
+        this.ordersController.deleteOrder(computeOrder);
+        this.ordersController.deleteOrder(volumeOrder);
```

Comment:
shouldn't we also check the object was actually removed?

Reply:
According to the test case, it expects no exception to be thrown, but I also think there should be this check, which in this case is to check if orders have been closed.

Reply:
I added checks to the test to monitor the open orders list.